### ISSUE TITLE
GetDocument throws NullReferenceException

### ISSUE THREAD

**Issue by s-KaiNet** [docusign/docusign-csharp-client#97]
I'm using nuget 2.0.1 version of the library and trying to download the file. Every time I got 
System.NullReferenceException: Object reference not set to an instance of an object.

```
at DocuSign.eSign.Client.ApiClient.Deserialize(IRestResponse response, Type type) in y:\Docusign\Github\DevCenter\SDKs\csharp\sdk\src\main\csharp\DocuSign\eSign\Client\ApiClient.cs:line 263 
   at DocuSign.eSign.Api.EnvelopesApi.GetDocumentWithHttpInfo(String accountId, String envelopeId, String documentId) in y:\Docusign\Github\DevCenter\SDKs\csharp\sdk\src\main\csharp\DocuSign\eSign\Api\EnvelopesApi.cs:line 4792 
   at DocuSign.eSign.Api.EnvelopesApi.GetDocument(String accountId, String envelopeId, String documentId) in y:\Docusign\Github\DevCenter\SDKs\csharp\sdk\src\main\csharp\DocuSign\eSign\Api\EnvelopesApi.cs:line 4716 
```

My code: 

```
var stream = envelopeApi.GetDocument(context.AccountId, context.Envelope.EnvelopeStatus.EnvelopeID, signedDocStatus.ID);
```

I'm absolutely sure that provided values are not null and valid. I've also created small console application to verify this problem and it's still exists.  
Do you have any ideas on that issue?

**Comment by s-KaiNet** [comment]
```
var filePath = String.IsNullOrEmpty(Configuration.TempFolderPath)
                        ? Path.GetTempPath()
                        : Configuration.TempFolderPath;
```  

`Configuration` is null.

**Comment by s-KaiNet** [comment]
Ok guys, I was able to narrow down the issue. I debug the solution and found the root case of the problem (once again, I'm talking about the solution included into the nuget package, v2.0.1, under `packages\DocuSign.eSign.dll.2.0.1\DocuSign.eSign.sln`) Hopefully you included the code into the nuget package. 
In all your samples (for example [this one](https://www.docusign.com/developer-center/recipes/request-a-signature-via-email)) you are using the following code for login: 
```csharp
// we set the api client in global config when we configured the client 
ApiClient apiClient = Configuration.Default.ApiClient;
string authHeader = "{\"Username\":\"" + usr + "\", \"Password\":\"" + pwd + "\", \"IntegratorKey\":\"" + INTEGRATOR_KEY + "\"}";
Configuration.Default.AddDefaultHeader("X-DocuSign-Authentication", authHeader);
```  
I'm also using this code for login. Now I want to download the document and got `NullReferenceException`, because inside `DocuSign.eSign.Client.ApiClient` property `Configuration` is null. 
Now take a closer look to the `Configuration` class (nuget `packages\DocuSign.eSign.dll.2.0.1\src\main\csharp\DocuSign\eSign\Client\`): 
```csharp
public static Configuration Default = new Configuration();
```  
Inside default `Configuration` constructor there is a call to construct `ApiClient` object: 
```csharp
ApiClient = Default != null ? Default.ApiClient : new ApiClient();
```  
and inside default `ApiClient` constructor: 
```csharp
public ApiClient()
{
	Configuration = Configuration.Default;
	RestClient = new RestClient("https://www.docusign.net/restapi");
}
``` 
but at this stage `Configuration.Default` will be null, because we are still at the line `public static Configuration Default = new Configuration();`. This is a screenshot with stack trace to understand the issue: 
![2016-05-23_1252](https://cloud.githubusercontent.com/assets/10658276/15467145/9ed18878-20e5-11e6-81e8-8a2c28587bf7.png)
![2016-05-23_1254](https://cloud.githubusercontent.com/assets/10658276/15467148/a2c2e22e-20e5-11e6-834d-cb469d3a192b.png)

That means that in the following code: 
```csharp
ApiClient apiClient = Configuration.Default.ApiClient;
```  
apiClient.Configuration **will be always null...**

workaround is to use another constructor: 
```csharp
var apiClient = new ApiClient(Configuration.Default);
Configuration.Default.ApiClient = apiClient;
```  

I'm not sure about nuget package, may be it's not the latest, and this issue is related to #96  but you definitely need to do smth.

**Comment by LarryKlugerDS** [comment]
Thank you for using DocuSign. I'm closing out old issues that have not had any recent updates. Please re-open if the issue is still current. Thank you.
